app/src/main/java/com/vero/tasky/agenda/data/worker/SaveEventWorker.kt

```diff
@@ -106,12 +117,25 @@ class SaveEventWorker @AssistedInject constructor(
         val createEventRequest = adapter.toJson(
             agendaItem.event.toCreateEventRequest(agendaItem.attendees)
         )
-        api.createEvent(
+        val eventDto = api.createEvent(
             createEventRequest = MultipartBody.Part.createFormData(
                 "create_event_request",
                 createEventRequest
             ),
             photos = multipartPhotos
         )
+        updatePhotosEntities(eventDto, agendaItem)
+    }
+
+    private suspend fun updatePhotosEntities(eventDto: EventDto, agendaItem: EventWithPhotosAndAttendees) {
+        supervisorScope {
+            val remotePhotoJobs = eventDto.photos.map {
+                launch { eventDao.insertRemotePhotoEntity(it.toPhotoEntity(agendaItem.event.id)) }
+            }
+            val localPhotoJobs = agendaItem.localPhotos.map {
+                launch { eventDao.deleteLocalPhotos(it) }
```

Comment:
You will always remove all the local photos here, right?
If so, the dao could have a "deleteAllLocalPhotos", and just call it once. Instead of once on every photo